runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Configuration.java

```diff
@@ -832,6 +835,53 @@ public void properties(
         }
     }
 
+    public Map<String, Object> asMap()
+    {
+        return new AbstractMap<>()
+        {
+            @Override
+            public Set<Entry<String, Object>> entrySet()
+            {
+                Set<Entry<String, Object>> entries = new HashSet<>();
+                properties(
+                    (k, v) ->
+                    {
+                        if (v != null)
+                        {
+                            entries.add(Map.entry(k, v));
+                        }
+                    },
+                    (k, v) ->
+                    {
+                        if (v != null)
+                        {
+                            entries.add(Map.entry(k, v));
+                        }
+                    }
+                );
+                return entries;
+            }
+
+            @Override
+            public Object get(
+                Object key)
+            {
+                if (!(key instanceof String))
+                {
+                    return null;
+                }
+                for (Entry<String, Object> e : entrySet())
+                {
+                    if (e.getKey().equals(key))
+                    {
+                        return e.getValue();
+                    }
+                }
+                return null;
+            }
```

Comment:
I think we should be able to do better than a linear search here.
```suggestion
                Object value = null;

                if (key instanceof String name)
                {
                    PropertyDef property = definition.properties.get(name);
                    if (property != null)
                    {
                        value = property.get(Configuration.this);
                    }
                }
                
                return value;
            }
```